pkg/cloudfabric-controller/controllerframework/controller_framework.go

```diff
@@ -441,26 +437,27 @@ func (c *ControllerBase) ReportHealth(client clientset.Interface) {
 	}
 }
 
-func generateControllerName(existedInstanceMap map[string]v1.ControllerInstance) string {
-	for {
-		uid := uuid.NewUUID()
-		name := fmt.Sprintf("%s-%v", controllerInstanceNamePrefix, uid)
-		_, isExist := existedInstanceMap[name]
-		if !isExist {
-			return name
-		}
+func generateControllerName(controllerType string, existedInstanceMap map[string]v1.ControllerInstance) string {
```

Comment:
The fatal errors in this func are not returned, so will never be handled. Is it by design?

Reply:
> The fatal errors in this func are not returned, so will never be handled. Is it by design?

Yes, there is no solution to recover from this error aside from restart workload controller manager.